sqlmesh/lsp/completions.py

```diff
@@ -26,38 +31,42 @@ def get_sql_completions(
     # Combine keywords - SQL keywords first, then file keywords
     all_keywords = list(sql_keywords) + list(file_keywords - sql_keywords)
 
+    models = list(get_models(context, file_uri))
     return AllModelsResponse(
-        models=list(get_models(context, file_uri)),
+        models=[m.name for m in models],
+        model_completions=models,
         keywords=all_keywords,
         macros=list(get_macros(context, file_uri)),
     )
 
 
-def get_models(context: t.Optional[LSPContext], file_uri: t.Optional[URI]) -> t.Set[str]:
+def get_models(
+    context: t.Optional[LSPContext], file_uri: t.Optional[URI]
+) -> t.List[ModelCompletion]:
     """
     Return a list of models for a given file.
 
     If there is no context, return an empty list.
     If there is a context, return a list of all models bar the ones the file itself defines.
     """
     if context is None:
-        return set()
+        return []
+
+    current_path = file_uri.to_path() if file_uri is not None else None
+
+    completions: t.List[ModelCompletion] = []
```

Comment:
im thinking whether it'd make sense to push that try except logic within `generate_markdown_description` since we silently suppress and return none in that case anyway. and then here to simplify this as :
```suggestion
    completions: t.List[ModelCompletion] = [
        ModelCompletion(name=model.name, description= generate_markdown_description(model))
        for model in context.context.models.values()
        if current_path is None or model._path != current_path
    ]
```